account_invoice_split/wizard/account_invoice_split.py

```diff
@@ -0,0 +1,192 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#     This file is part of account_invoice_split,
+#     an Odoo module.
+#
+#     Copyright (c) 2015 ACSONE SA/NV (<http://acsone.eu>)
+#
+#     account_invoice_split is free software:
+#     you can redistribute it and/or modify it under the terms of the GNU
+#     Affero General Public License as published by the Free Software
+#     Foundation,either version 3 of the License, or (at your option) any
+#     later version.
+#
+#     account_invoice_split is distributed
+#     in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+#     even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+#     PURPOSE.  See the GNU Affero General Public License for more details.
+#
+#     You should have received a copy of the GNU Affero General Public License
+#     along with account_invoice_split.
+#     If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api, exceptions, _
+import openerp.addons.decimal_precision as dp
+
+FIELD_MAPPING = [('id', 'origin_invoice_line_id'),
+                 ('product_id', 'product_id'),
+                 ('name', 'name'),
+                 ('quantity', 'total_quantity')]
+
+
+class AccountInvoiceSplit(models.TransientModel):
+    _name = 'account.invoice.split'
+
+    line_ids = fields.One2many(
+        comodel_name='account.invoice.split.line', inverse_name='wizard_id',
+        string='Invoice lines to Split')
+
+    @api.model
+    def _check_condition(self):
+        active_ids = self.env.context.get('active_ids')
+        assert len(active_ids) == 1
+        invoice = self.env['account.invoice'].browse(active_ids)[0]
+        if invoice.state != 'draft':
+            raise exceptions.Warning(_('Invoice must be draft not %s') %
+                                     (invoice.state,))
+
+    @api.model
+    def default_get(self, fields_list):
+        self._check_condition()
+        res = super(AccountInvoiceSplit, self).default_get(self)
+        active_ids = self.env.context.get('active_ids')
+        assert len(active_ids) == 1
+        invoice = self.env['account.invoice'].browse(active_ids)[0]
+        split_lines = []
+        for line in invoice.invoice_line:
+            current_line_values = {
+                'origin_invoice_line_id': line.id,
+                'product_id': line.product_id.id,
+                'name': line.name,
+                'total_quantity': line.quantity,
+            }
+            split_lines.append((0, 0, current_line_values))
+        res['line_ids'] = split_lines
+        return res
+
+    @api.model
+    def _get_invoice_values(self, invoice):
```

Comment:
Why don't you just copy the invoice record and change what needed (and the same for lines)? Probably this way you avoid some glue modules